### ISSUE TITLE
Support for API endpoint to trigger CD jobs

### ISSUE THREAD

**Issue by Jsgoller1** [Shippable/support#3401]
***tl;dr - I want to automate running CD jobs, found an undocumented endpoint that appears to work, is this acceptable y/n?***

### Description of your issue:
We use Shippable for both CI and CD. We have a Release job defined for one of our containers in a `shippable.jobs.yml` file:
```
- name: crm-zp-release
  type: release
  flags:
  - "service: crm"
  - "env: zp"
  steps:
  - IN: crm-version
    switch: off
  - IN: crm-zs-deploy
    switch: off
  - TASK: managed
    bump: minor
```
This results in a Job in the Single Pane of Glass (SPOG) interface with a "play" button on it that allows it to be run manually by clicking on it. I would like to run this from the API without looking at the SPOG (so that I can automate running the Release with `curl`/`make` or Ansible). I have found that clicking on the play button causes a POST request to be sent to `https://api.shippable.com/resources/1234/triggerNewBuildRequest`, where `1234` is the `ID` value in the pop-up box (above the play button), but I don't see any documentation for the `/resources` endpoint in the API documentation. Is there an easier / better way to run CD jobs or is this functionality not supported in Shippable? I can't find any other documentation for running CD jobs from the API.

**Comment by avinci** [comment]
this is the right way to do it. We are documenting all our APIs through swagger by end of next week and you should see this documented. 

Every action from the UI has an API and we have not done a good job to document it. We have take this up i.e. improve docs and help files as our primary focus over the next few weeks to avoid users like yourself try and hack it.

**Comment by Jsgoller1** [comment]
Thanks for the quick turnaround on this issue, @avinci!

**Comment by BlueDragonX** [comment]
@avinci is there any way via the API to look up jobs by name?

**Comment by himanshu0503** [comment]
@BlueDragonX : You can query this by using the route
```
https://api.shippable.com/resources?names=resource1Name,resource2Name
```

**Comment by BlueDragonX** [comment]
Thanks!
